### PR TITLE
Web console support for  JenkinsPipeline builds

### PR DESCRIPTION
**Pull request by spadgett**
https://trello.com/c/KMyd8ETi

Based on https://github.com/openshift/origin/pull/7949

View Jenkins Pipeline build:

openshift web console

Edit Jenkins Pipeline build:

edit-jenkins-bc

/cc @jwforres @bparees

### CODE REVIEW

**Comment by jwforres** [comment]
yeah our previous discussion about the read-only view looking just like the
editable view still seems like a problem.  or did you do the thing where
you can edit it and a Save button will pop up?

**Comment by spadgett** [comment]
No, I can change it. Probably want to change it for Dockerfile, too.

**Comment by spadgett** [comment]
More readonly-y ?

openshift_web_console

**Comment by jwforres** [comment]
Yeah better, though looks like it could use some padding

**Comment by bparees** [comment]
jenksfiles are likely going to be really long in real world use, so i'm wondering if we even want to display it in the readonly view.  or at least have it hidden by default or something?  Otherwise they're going to scroll triggers off the page (unless triggers is moved above) and in general just look like a wall of text.

**Comment by jwforres** [comment]
We set a max height on the editor and it internally scrolls.

**Comment by bparees** [comment]
by editor you mean the read-only editor too?

**Comment by jwforres** [comment]
Yes

**Comment by bparees** [comment]
Ok. concerns addressed.

File: assets/app/views/edit/build-config.html [hunk 1]
```diff
@@ -141,6 +142,54 @@
...
+                                     id="jenkinsfilePath"
+                                     name="jenkinsfilePath"
+                                     type="text"
+                                     ng-model="updatedBuildConfig.spec.strategy.jenkinsPipelineStrategy.jenkinsfilePath"
+                                     autocorrect="off"
+                                     autocapitalize="off"
+                                     spellcheck="false"
+                                     aria-describedby="jenkinsfile-path-help">
+                              <div class="help-block" id="jenkinsfile-path-help">
+                                Optional path to the Jenkinsfile realtive to the root of the source
```
Comment (jwforres) [inline assets/app/views/edit/build-config.html]: relative to root of the repository or to the context-dir ?  @bparees ?

**Comment by jwforres** [comment]
just the one question otherwise i think this is ready to be merged

File: assets/app/views/edit/build-config.html (hunk 1 above)
Comment (bparees) [inline assets/app/views/edit/build-config.html]: s/realative/relative/

and relative to the context dir, good catch @jwforres 

https://github.com/openshift/origin/pull/7949/files#diff-43dd40b4c2b85b388a3d9116729032e4R416

Comment (jimmidyson) [inline assets/app/views/edit/build-config.html]: This is already sorted in the PR - please pull in latest changes.

**Comment by spadgett** [comment]
@jwforres updated

**Comment by jwforres** [comment]
Merge conflict

**Comment by spadgett** [comment]
Rebased.

**Pull request merged**